### PR TITLE
Apply partial set (applyOwnerChange()) in setOwner()

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Apply partial set (applyOwnerChange()) in setOwner()
- Fix partial set application condition of setFunction()

Related issues: https://github.com/ainblockchain/ain-blockchain/issues/409

### CODE REVIEW

File: db/state-util.js [hunk 1]
```diff
@@ -336,73 +336,143 @@ function isValidConfigTreeRecursive(stateTree, path, configLabel, stateConfigVal
...
+}
+
+/**
+ * Returns a new function tree created by applying the function change to
+ * the current function tree.
+ * @param {Object} curFuncTree current function tree (to be modified by this function)
+ * @param {Object} functionChange function change
+ */
+function applyFunctionChange(curFuncTree, functionChange) {
+  // NoTE(platfowner): Partial set is applied only when the current function tree has
```
[the remark this answers is not in the document]
Comment (platfowner) [reply db/state-util.js]: Fixed. Thanks!

File: db/state-util.js [hunk 2]
```diff
@@ -336,73 +336,143 @@ function isValidConfigTreeRecursive(stateTree, path, configLabel, stateConfigVal
...
- * Returns a new function created by applying the function change to the current function.
- * @param {Object} curFunction current function (to be modified and returned by this function)
- * @param {Object} functionChange function change
+ * Returns whether the given state tree object has the given config label as a property.
  */
-function applyFunctionChange(curFunction, functionChange) {
-  if (curFunction === null) {
-    // Just write the function change.
-    return functionChange;
+function hasConfigObj(stateTreeObj, configLabel) {
```
[the remark this answers is not in the document]
Comment (platfowner) [reply db/state-util.js]: Just added null check to cover .write config (we don't have any plan for the use case yet though). I think there won't be any ambiguity issue as `isValidRuleConfig()` does not allow null value. Thanks for the comment!

**Comment by platfowner** [comment]
Addressed all comments and fixed a bug. PTAL @liayoo

File: db/state-util.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (liayoo) [reply db/state-util.js]: I see! Thanks for the clarification! 👍

**Comment by liayoo** [comment]
LGTM, thanks!

**Review by liayoo — APPROVED** [review]

**Comment by platfowner** [comment]
Thanks for the fast review! Now merging..

**Pull request merged**